### PR TITLE
Add experimental riscv64 mention and support to get-dart

### PR DESCRIPTION
**Pull request by parlough**
Fixes https://github.com/dart-lang/site-www/issues/3958 

Makes progress towards https://github.com/dart-lang/site-www/issues/3906

**Staged Get Dart:** https://dart-dev--pr3959-feature-riscv64-arch-ps1itypv.web.app/get-dart

**Staged archive:** https://dart-dev--pr3959-feature-riscv64-arch-ps1itypv.web.app/get-dart/archive

### CODE REVIEW

**Comment by parlough** [comment]
@mit-mit Can you take a look at this now that riscv builds are in the latest dev build? Thanks!

**Review by domesticmouse — APPROVED** [review]

File: src/get-dart/index.md [hunk 1]
```diff
@@ -61,7 +61,8 @@ The Dart SDK is supported on Windows, Linux, and macOS.
 ### Linux
 
 * **Supported versions:** [Debian stable][] and [Ubuntu LTS][] under standard support.
-* **Supported architectures:** x64, ia32, arm, arm64.
+* **Supported architectures:** x64, ia32, arm64, arm, _riscv64_.<br>
+  Support for riscv4 is in preview, and is available only in the dev channel.
```
Comment (mit-mit) [inline src/get-dart/index.md:65]: s/for riscv4/for RISC-V/

File: src/get-dart/dart_sdk_archive/lib/src/util.dart [hunk 2]
```diff
@@ -42,6 +42,7 @@ const Map<String, String> archiveMap = {
   'ARM64': 'arm64',
   'ARMv7': 'arm',
   'ARMv8 (ARM64)': 'arm64',
+  'riscv64': 'riscv64',
```
Comment (mit-mit) [inline src/get-dart/dart_sdk_archive/lib/src/util.dart:45]: s/'riscv64':/'RISC-V (RV64GC)':/

File: src/get-dart/dart_sdk_archive/lib/src/util.dart [hunk 3]
```diff
@@ -66,6 +67,7 @@ const Map<String, List<PlatformVariant>> platforms = {
     PlatformVariant('ia32', ['Dart SDK']),
     PlatformVariant('ARMv8 (ARM64)', ['Dart SDK']),
     PlatformVariant('ARMv7', ['Dart SDK']),
+    PlatformVariant('riscv64', ['Dart SDK']),
```
Comment (mit-mit) [inline src/get-dart/dart_sdk_archive/lib/src/util.dart:70]: s/'riscv64':/'RISC-V (RV64GC)':/

File: src/get-dart/index.md [hunk 4]
```diff
@@ -61,7 +61,8 @@ The Dart SDK is supported on Windows, Linux, and macOS.
 ### Linux
 
 * **Supported versions:** [Debian stable][] and [Ubuntu LTS][] under standard support.
-* **Supported architectures:** x64, ia32, arm, arm64.
+* **Supported architectures:** x64, ia32, arm64, arm, _riscv64_.<br>
```
Comment (mit-mit) [inline src/get-dart/index.md:64]: Let's change this to:

```
**Supported architectures:** x64, IA32, ARM64, ARM, RISC-V (RV64GC).<br>
```

**Review by mit-mit — CHANGES_REQUESTED** [review]
Thanks! Some detailed changes requested

**Review by mit-mit — APPROVED** [review]
Thanks!

**Pull request merged**